eslint.config.mjs

```diff
@@ -1,18 +1,23 @@
 // @ts-check
-"use strict";
 
-const configInternal = require("@babel/eslint-config-internal");
-const pluginImport = require("eslint-plugin-import");
-const pluginN = require("eslint-plugin-n");
-const pluginPrettier = require("eslint-plugin-prettier");
-const pluginBabelDevelopment = require("@babel/eslint-plugin-development");
-const pluginBabelDevelopmentInternal = require("@babel/eslint-plugin-development-internal");
-const typescriptEslint = require("typescript-eslint");
+import configInternal from "@babel/eslint-config-internal";
+// @ts-expect-error no types
+import pluginImport from "eslint-plugin-import";
+import pluginN from "eslint-plugin-n";
+import pluginPrettier from "eslint-plugin-prettier";
+// @ts-expect-error no types
+import pluginBabelDevelopment from "@babel/eslint-plugin-development";
+// @ts-expect-error no types
+import pluginBabelDevelopmentInternal from "@babel/eslint-plugin-development-internal";
+import typescriptEslint from "typescript-eslint";
+import { createRequire } from "module";
 
-const { FlatCompat } = require("@eslint/eslintrc");
+import { FlatCompat } from "@eslint/eslintrc";
+
+const require = createRequire(import.meta.url);
 
 const compat = new FlatCompat({
-  baseDirectory: __dirname,
+  baseDirectory: import.meta.dirname,
```

Comment:
Unfortunately, vscode bundled an old version of Node.js[^1], so this will not work within vscode.
https://github.com/microsoft/vscode-eslint/issues/1851

[^1]: https://github.com/ewanharris/vscode-versions

Reply:
```suggestion
  baseDirectory: path.dirname(fileURLToPath(import.meta.url)),
```

or

```suggestion
  baseDirectory: import.meta.dirname ?? path.dirname(fileURLToPath(import.meta.url)),
```

Reply:
I checked out this branch locally and it seems that vscode works for me under the default `vscode-eslint` config. There is also an `eslint.runtime` config item where you can provide a node binary to run `vscode-eslint`.

Reply:
@JLHwung Thanks for explanation. However, I discovered that vscode-eslint will use the bundled Node.js if `eslint.runtime` is not specified. Should we include it to `.vscode/settings.example.json` as well?

Reply:
I used `__dirname` here.